### PR TITLE
MWPW-158911: Unable to scroll page if the filters modal closed by click outside of the modal

### PR DESCRIPTION
**Pull request by mirafedas**
MWPW-158911:
In `merch-sidenav.js`, when the filters modal is closed by clicking outside of it (rather than using the 'Close' button), the overlay emits a close event. On this event, we need to remove the merch-modal class; otherwise, the page will remain unscrollable.
!!! The related Milo PR has to be merged first for this feature to work: https://github.com/adobecom/milo/pull/2933

MWPW-159427:
On iOS version 15 and lower there is a different way of managing the vertical stacking context, which puts the modal below the merch card. Adding z-index on the merch-sidenav resolves this problem.
Related Milo PR: https://github.com/adobecom/milo/pull/2994

Resolves: [MWPW-158911](https://jira.corp.adobe.com/browse/MWPW-158911) and [MWPW-159427](https://jira.corp.adobe.com/browse/MWPW-159427)

**Test URLs for MWPW-158911:**
- Before: https://main--cc--adobecom.hlx.live/products/catalog?martech=off
- After: https://mwpw-158911-modal-scroll --cc--adobecom.hlx.live/products/catalog?martech=off&milolibs=mwpw-158911-modal-scroll--milo--mirafedas
(added a space after the 'scroll' to make this link invisible for PSI check)

**Test URLs for MWPW-159427:**
- Before: https://main--cc-–adobecom.hlx.live/products/catalog?martech=off
- After: https://mwpw-159427-ipad-issues --cc-–adobecom.hlx.live/products/catalog?martech=off (added a space to hide the link from PSI check)
- 
For PSI check:
https://mwpw-158911-modal-scroll--cc--adobecom.hlx.live/products/animate/free-trial-download?martech=off&georouting=off

### CODE REVIEW

**Comment by yesil** [comment]
@mirafedas can you link the Milo PR?

File: README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ The Franklin based project for creative cloud pages on www.adobe.com. Based off
 ## Code merge process
 https://github.com/adobecom/cc/wiki
 
+
```
Comment (Axelcureno) [inline README.md:7]: What's the change here? I only see changes on minified dependency, is there any src change?

Comment (mirafedas) [reply README.md]: It got there by mistake, thanks for noticing, I removed it

**Review by yesil — CHANGES_REQUESTED** [review]
@mirafedas why this file is needed in CC? It is in Milo already.
can you try to remove it from the CC?
Only the merch-sidenav block remains in CC and later it can moved back to Milo with Plans migration.
cc: @npeltier

**Comment by Roycethan** [comment]
@mirafedas  https://github.com/adobecom/milo/pull/2933 is verified, plz get enough approvals for this PR thanks

**Comment by mirafedas** [comment]
@yesil I added the change to use the `merch-sidenav.js` directly from Milo, could you please review?

**Review by yesil — APPROVED** [review]

**Review by npeltier — APPROVED** [review]

**Pull request merged**